lib/susedistribution.pm

```diff
@@ -571,10 +571,33 @@ sub init_consoles {
             my $packed_ip = gethostbyname($s390_guest_fqdn);
             die "Failed to get host by name for '$s390_guest_fqdn' (on " . hostname . ")" unless $packed_ip;
             my $s390_guest_ip = inet_ntoa($packed_ip);
-            my $host_ip = get_var("AGAMA") ? 'hostip' : 'HostIP';
-            my $host_name = get_var("AGAMA") ? 'hostname' : 'Hostname';
-            $s390_params .= " $host_ip" . "=${s390_guest_ip}/${s390_guest_subnetmask}";
-            $s390_params .= " $host_name" . "=${s390_guest_hostname}";
+            if(get_var("AGAMA")) {
+                my @split = split(/\./, get_required_var("REPO_HOST"));
+                my $s390_gateway_ip = join(".", @split[0], @split[1], @split[2], "254");
+
+                my ($hostname) = split(/\./, get_required_var("ZVM_GUEST"));
+
+                if(is_sle()){
+                    @split = split(/\./, get_required_var("ZVM_GUEST"));
+                    my $s390_domain = join( ".", @split[1], @split[2], @split[3], @split[4]);
```

Comment:
the use of index is never a good idea, makes the code harder to maintain, another approach would be to get a list with the values separated by dots and then pop from the list the first element and join the remaining, in that way you can use it for both cases.

Reply:
6742aaafb